jena-shex/src/test/java/org/apache/jena/shex/runner/ShexTests.java

```diff
@@ -329,10 +325,41 @@ private static boolean runTestExclusionsInclusions(ManifestEntry entry) {
                 //System.err.println("Skipping:  "+fragment);
                 return false;
             }
+
+            List<Resource> traits = extractTraits(entry);
+            if (traits != null) {
+                List<Resource> excludedBecause = traits.stream().filter(excl -> excludeTraits.contains(excl)).collect(Collectors.toList());
```

Comment:
Think you can simplify the whole interior of this if block to just `return !traits.stream().anyMatch(excl -> excludeTraits.contains(excl));`

Reply:
I compose the `excludedBecause` because (because because) I used it to see why tests were excluded. It could be nice to have a command-line arg to log it but at least this way you can println it to chase down mysteries.